bin/compare_threshold.py

```diff
@@ -1,138 +1,55 @@
-    parser.add_argument("--mosdepth_global_summary", "-s", required=True)
-    parser.add_argument("--mosdepth_target_regions_bed", "-b", required=True)
-    parser.add_argument("--thresholds", "-t", required=True)
-    parser.add_argument("--fastp_json", "-f", required=True, nargs="+", help="fastp json file(s)")
-    parser.add_argument("--sample_id", "-i", required=True)
-    parser.add_argument("--labDataName", "-n", required=True)
-    parser.add_argument("--libraryType", "-l", required=True)
-    parser.add_argument("--sequenceSubtype", "-a", required=True)
-    parser.add_argument("--genomicStudySubtype", "-g", required=True)
-    parser.add_argument("--output", "-o", required=True)
-    return parser.parse_args(args)
-
-def main(args=None):
-    args = parse_args(args)
-
-    # Read thresholds JSON
-    with open(args.thresholds, "r") as f:
-        thresholds_data = json.load(f)
-
-    thresholds = None
-    for item in thresholds_data:
-        if (
-            item["libraryType"] == args.libraryType
-            and item["sequenceSubtype"] == args.sequenceSubtype
-            and item["genomicStudySubtype"] == args.genomicStudySubtype
-        ):
-            thresholds = item["thresholds"]
-            break
-
-    if thresholds is None:
-        raise ValueError(
-            "No matching thresholds found for meta values: "
-            + args.libraryType + ", " + args.sequenceSubtype + ", " + args.genomicStudySubtype
-        )
 
-    ### Collect all statistics
+def main(args: argparse.Namespace):
+    threshold_defs = json.loads(
+        (
+            importlib.resources.files(grz_pydantic_models.resources) / "thresholds.json"
+        ).read_text()
+    )
+    keys2threshold = {}
+    for threshold_def in threshold_defs:
+        key = (
+            threshold_def["libraryType"],
+            threshold_def["sequenceSubtype"],
+            threshold_def["genomicStudySubtype"],
+        )
+        if key in keys2threshold:
+            raise ValueError(
+                "Thresholds definition file contains duplicate definitions"
+            )
+        keys2threshold[key] = threshold_def
 
-    # --- Determine 'meanDepthOfCoverage' ---
+    thresholds = keys2threshold[
+        (args.libraryType, args.sequenceSubtype, args.genomicStudySubtype)
+    ]["thresholds"]
 
-    # Required mean depth of coverage to pass the validation
-    mean_depth_of_converage_required = float(thresholds["meanDepthOfCoverage"])
-    # Read mosdepth summary file
-    df = pd.read_csv(args.mosdepth_global_summary, sep="\t")
-    row_name = "total_region" if args.libraryType in ["panel", "wes", "panel_lr", "wes_lr"] else "total"
-    mean_depth_of_coverage = df.loc[df["chrom"] == row_name, "mean"].item()
+    # threshold - mean depth of coverage
+    mosdepth_summary_df = pd.read_csv(
+        args.mosdepth_global_summary, sep="\t", header=0, index_col="chrom"
+    )
+    mean_depth_of_coverage = mosdepth_summary_df.loc["total_region", "mean"]
```

Comment:
I think this is one of the most important changes in this PR (besides the % deviation calculation). We should always use `total_region` from mosdepth because we always pass a BED file, just in WGS case it is 400 representative genes.

Reply:
TODO: Add documentation that clarifies the 400 gene set only applies to target region coverage.

Reply:
It seems [docs/details.md](https://github.com/BfArM-MVH/GRZ_QC_Workflow/blob/main/docs/details.md#mosdepth) already documents this and I just need to learn how to read better :smile: